From abf1da3f169cd6d28354aa6aab539f0da1d64ea9 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:49 -0500
Subject: [drm] radeon: fix active cu count for SI and CIK

Message-id: <1418852380-13061-39-git-send-email-rclark@redhat.com>
Patchwork-id: 102177
O-Subject: [RHEL7 drm 38/89] drm/radeon: fix active cu count for SI and CIK
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit d78a30804a0d47f16f17d99cc74021030503efbf

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Tue Aug 19 11:54:15 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:48 2014 -0500

    drm/radeon: fix active cu count for SI and CIK

    commit 6101b3ae94b4f266456308824e9ca4eab1235d1a upstream.

    This fixes the CU count reported to userspace for
    OpenCL.

    bug:
    https://bugzilla.kernel.org/show_bug.cgi?id=82581

    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Reviewed-by: Michel Danzer <michel.daenzer@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/cik.c b/drivers/gpu/drm/radeon/cik.c
index feb526c..a5ce5e6 100644
--- a/drivers/gpu/drm/radeon/cik.c
+++ b/drivers/gpu/drm/radeon/cik.c
@@ -3259,7 +3259,7 @@ static void cik_gpu_init(struct radeon_device *rdev)
  u32 mc_shared_chmap, mc_arb_ramcfg;
  u32 hdp_host_path_cntl;
  u32 tmp;
- int i, j, k;
+ int i, j;
 
  switch (rdev->family) {
  case CHIP_BONAIRE:
@@ -3451,10 +3451,8 @@ static void cik_gpu_init(struct radeon_device *rdev)
 
  for (i = 0; i < rdev->config.cik.max_shader_engines; i++) {
   for (j = 0; j < rdev->config.cik.max_sh_per_se; j++) {
-   for (k = 0; k < rdev->config.cik.max_cu_per_sh; k++) {
-    rdev->config.cik.active_cus +=
-     hweight32(cik_get_cu_active_bitmap(rdev, i, j));
-   }
+   rdev->config.cik.active_cus +=
+    hweight32(cik_get_cu_active_bitmap(rdev, i, j));
   }
  }
 
diff --git a/drivers/gpu/drm/radeon/si.c b/drivers/gpu/drm/radeon/si.c
index f87d82a..238ff93 100644
--- a/drivers/gpu/drm/radeon/si.c
+++ b/drivers/gpu/drm/radeon/si.c
@@ -2901,7 +2901,7 @@ static void si_gpu_init(struct radeon_device *rdev)
  u32 sx_debug_1;
  u32 hdp_host_path_cntl;
  u32 tmp;
- int i, j, k;
+ int i, j;
 
  switch (rdev->family) {
  case CHIP_TAHITI:
@@ -3101,10 +3101,8 @@ static void si_gpu_init(struct radeon_device *rdev)
 
  for (i = 0; i < rdev->config.si.max_shader_engines; i++) {
   for (j = 0; j < rdev->config.si.max_sh_per_se; j++) {
-   for (k = 0; k < rdev->config.si.max_cu_per_sh; k++) {
-    rdev->config.si.active_cus +=
-     hweight32(si_get_cu_active_bitmap(rdev, i, j));
-   }
+   rdev->config.si.active_cus +=
+    hweight32(si_get_cu_active_bitmap(rdev, i, j));
   }
  }
 
-- 
1.7.1